fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -0,0 +1,109 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.connector.jdbc;
+
+import com.google.common.collect.ImmutableList;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.JDBCResource;
+import com.starrocks.catalog.JDBCTable;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.DdlException;
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.external.HiveMetaStoreTableUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class JDBCMetadata implements ConnectorMetadata {
+
+    private static Logger LOG = LogManager.getLogger(JDBCMetadata.class);
+
+    private Map<String, String> properties;
+    private JDBCSchemaResolver schemaResolver;
+
+    public JDBCMetadata(Map<String, String> properties) throws DdlException {
+        this.properties = properties;
+        try {
+            Class.forName(properties.get(JDBCResource.DRIVER_CLASS));
+        } catch (ClassNotFoundException e) {
+            LOG.warn(e.getMessage());
+            throw new DdlException("doesn't find class: " + e.getMessage());
+        }
+        if (properties.get(JDBCResource.DRIVER_CLASS).toLowerCase().contains("mysql")) {
+            schemaResolver = new MysqlSchemaResolver();
+        } else {
+            LOG.warn("not support yet");
+            throw new DdlException("not support yet");
+        }
+    }
+
+    private Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(properties.get(JDBCResource.URI),
+                properties.get(JDBCResource.USER), properties.get(JDBCResource.PASSWORD));
+    }
+
+    @Override
+    public List<String> listDbNames() throws DdlException {
+        try (Connection connection = getConnection()) {
+            return schemaResolver.listSchemas(connection).stream()
+                    .map(String::toLowerCase)
+                    .collect(Collectors.toList());
+        } catch (SQLException e) {
+            throw new DdlException(e.getMessage());
+        }
+    }
+
+    @Override
+    public Database getDb(String name) {
+        try {
+            if (listDbNames().contains(name)) {
+                return new Database(HiveMetaStoreTableUtils.connectorDbIdIdGenerator.getNextId().asInt(), name);
```

Comment:
Why do we need HiveMetaStoreTableUtils here?

Reply:
we use the databaseidgenerator and tableidgenerator so that all the connectors will have continuous id.

Reply:
If this id is not useful, how about give a same id?
`HiveMetaStoreTableUtils.connectorDbIdIdGenerator.getNextId()` this will return different id for the same database.